sstables/compaction.cc

```diff
@@ -462,10 +462,14 @@ class compaction {
         };
     }
 
-    virtual std::function<bool(const dht::decorated_key&)> filter_func() const {
-        return [] (const dht::decorated_key&) {
+    virtual bool filter_func(const dht::decorated_key&) const {
```

Comment:
The name isn't really meaningful. Maybe 'consider_partition()`, or something similar that expresses the purpose of this function, which is whether the compaction even considers that partition for the output or not.

Reply:
I agree. It is a bit out of scope of this patchset but I can squeeze that in the next version while at it.

Reply:
@denesb now with the forwarding constructor there is no change in this code so it's hard to justify this rename as part of this patchset.  For the sake of backportability, let's keep it minimal and I'll send a followup PR with the cosmetic improvements (for master, not to be backported).

Reply:
hmm, unfortunately the `filter_func` signature does need to change apparently.  So I'll include the renaming patch after all.